### DISCUSSION TITLE
Why does PartialInputStream.Seek not return a relative value?

### DISCUSSION THREAD

**Discussion by mrtop** [icsharpcode/SharpZipLib#794] [Q&A]
PartialInputStream.Position  return The current position within the stream.
but PartialInputStream.Seek return The current position within the baseStream.
why?

![image](https://user-images.githubusercontent.com/10303227/199933243-dd8e4577-4344-4ad9-b665-dfc76d9c3210.png)

It should not be  readPos_ - start ?

**Comment by piksel** [comment]
I guess you could argue either would be the "correct" return value here. My guess is that it's because `Seek` would be used by something else that needs the non-relative offset.

Ideally, it feels like that nothing from the base stream should leak to the outer API, and perhaps this was even a mistake. Unfortunately, changing this would be a breaking change, but the docs should probably mention this.

**Reply by mrtop** [reply]
So normally it should be "return readPos_ - start_"
It can be changed in my own project without affecting its other functions
